lib/dry/component/container.rb

```diff
@@ -123,6 +123,8 @@ def self.require(*paths)
       end
 
       def self.load_component(key)
+        return if key?(key)
```

Comment:
I like the convention that a method that changes an object (container class
in this case) always returns self. WDYT?

On 31 July 2016 at 14:31:38, Tim Riley (notifications@github.com) wrote:

> In lib/dry/component/container.rb
> <https://github.com/dry-rb/dry-component/pull/24#discussion_r72903496>:
>
> > @@ -123,6 +123,8 @@ def self.require(*paths)
> >        end
> >
> >        def self.load_component(key)
> > +        return if key?(key)
>
> Oh, right :) We don't operate on .load_component's return value anywhere,
> so I think a nil return in this case would be fine.
>
> Perhaps I can make this PR more robust by adding a small yardoc snippet to
> the top making it clear this is a method with a "void" return?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/dry-rb/dry-component/pull/24/files/d94fca5226cd8093bad13cf985a34762aa1037b9#r72903496>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAEKm2EgKC-gXpl1cT_kzVwrXfWkG1Dks5qbJWqgaJpZM4JY8xo>
> .
>